### PR TITLE
DShot & ESC sensor support for SpeedyBee F4

### PR DESCRIPTION
**Pull request by seriousrob**
Hi guys,

changed some files for the SpeedyBee F4 to add DShot, ESC sensor and SoftSerial support.

Changed timers in the config.c
Recreated the whole target.c file for better readability. Added if-else for the SoftSerial targets
Created target files for using 1 or 2 SoftSerials.

SoftSerial 1 on S5 RX and S7 TX
SoftSerial 2 on S6 RX and DAC TX

Did some test flights this weekend and had to make a late change to get the Blackbox working. Tested all files and targets up to DShot 600, since 1200 isn't recommended by iNav. DShot 1200 will reboot some of the ESC after throttle cuts.

Experienced no issues so far on my setup:
FC: SpeedyBee F4 V2
ESC: 4xHAKRC32 35A BLHeli_32
GPS: DJI NAZA GPS
Baro: MS5611
VTX: EACHINE TX805 with SA
RX: Jumper R1 with S.Port on SoftSerial 1

regards
Robert

### CODE REVIEW

File: src/main/target/SPEEDYBEEF4/target.h [hunk 1]
```diff
@@ -138,24 +155,22 @@
...
-#define SENSORS_SET (SENSOR_ACC|SENSOR_MAG|SENSOR_BARO)
+/*** LED STRIP ***/
 
-#define DEFAULT_RX_TYPE         RX_TYPE_PPM
-#define DISABLE_RX_PWM_FEATURE
+/*** Default settings ***/
 #define DEFAULT_FEATURES        (FEATURE_TX_PROF_SEL | FEATURE_CURRENT_METER | FEATURE_BLACKBOX | FEATURE_VBAT | FEATURE_OSD)
+#define DEFAULT_RX_TYPE         RX_TYPE_SERIAL
+#define SERIALRX_PROVIDER       SERIALRX_SBUS
+#define CURRENT_METER_SCALE     166
```
Comment (digitalentity) [inline src/main/target/SPEEDYBEEF4/target.h:164]: Previously we had default value of 302, why this is changed?

Comment (seriousrob) [reply src/main/target/SPEEDYBEEF4/target.h:164]: The old value was way too high atleast for me. I've flown it really often on the stock 2.4.0 firmware and had around 2500mAh drawn, recharged about 4000mAh. On a brand new Turnigy NanoTech 4S LiPo. Checked the behaviour with other LiPos in my bag and the values i calculated for me are the same. That's why i changed it.

Comment (seriousrob) [reply src/main/target/SPEEDYBEEF4/target.h:164]: The old value was way too high atleast for me. I've flown it really often on the stock 2.4.0 firmware and had around 2500mAh drawn, recharged about 4000mAh. On a brand new Turnigy NanoTech 4S LiPo. Checked the behaviour with other LiPos in my bag and the values i calculated for me are the same. That's why i changed it.

**Review by digitalentity — APPROVED** [review]
LGTM

**Comment by DzikuVx** [comment]
@seriousrob checks seems to be hung. Could you please rebase od "develpment" to retrigger jobs?

**Comment by seriousrob** [comment]
@DzikuVx 
i'm farely new to the whole github ecosystem and am not quite sure what i need to do right now. I edited the title, selected base:development again and saved it. I hope that's what you mean otherwise you may need to describe more precisely, what i have to do.

ty

**Comment by digitalentity** [comment]
I've force-merged this. Let's see if it gets through the RC-cycle

**Comment by seriousrob** [comment]
@digitalentity Thanks for force merging this. I can see in the source code of 2.5.0 RC2 that the files are integrated but unfortunately i'm unable to select the added SoftSerial Targets via the new configurator.

Is there anything i have to add to change that?

**Pull request merged**